packages/integration/src/gitlab/GitLabIntegration.ts

```diff
@@ -62,6 +81,92 @@ export class GitLabIntegration implements ScmIntegration {
   resolveEditUrl(url: string): string {
     return replaceGitLabUrlType(url, 'edit');
   }
+
+  fetch(input: RequestInfo | URL, init?: RequestInit): Promise<Response> {
+    return this.fetchImpl(input, init);
+  }
+
+  private createFetchStrategy(): FetchFunction {
+    let fetchFn: FetchFunction = async (url, options) => {
+      return fetch(url, { ...options, mode: 'same-origin' });
+    };
+
+    // Early return for plain fetch (no enhancements needed)
+    if (
+      this.integrationConfig.limitPerMinute <= 0 &&
+      this.integrationConfig.maxRetries <= 0
+    ) {
+      return fetchFn;
+    }
+
+    // Apply retry wrapper if configured
+    if (this.integrationConfig.maxRetries > 0) {
+      fetchFn = this.withRetry(fetchFn);
+    }
+
+    // Apply throttling wrapper if configured
+    if (this.integrationConfig.limitPerMinute > 0) {
+      fetchFn = pThrottle({
+        limit: this.integrationConfig.limitPerMinute,
+        interval: 60_000,
+      })(fetchFn);
+    }
+
+    return fetchFn;
+  }
+
+  private withRetry(fetchFn: FetchFunction): FetchFunction {
+    return async (url, options) => {
+      let retries = 0;
+
+      while (retries <= this.integrationConfig.maxRetries) {
+        this.queue.push(() => {});
+
+        const response = await fetchFn(url, options);
+
+        if (
+          !this.integrationConfig.retryStatusCodes.includes(response.status) ||
+          retries >= this.integrationConfig.maxRetries
+        ) {
+          this.queue.pop();
+          return response;
+        }
+
+        if (response.status === 429) {
+          this.counter429?.add(1);
+        }
+
+        retries++;
+
+        // Calculate delay from Retry-After header or use exponential backoff
+        // Note: project, groups and users APIs may not respond with Retry-After header
+        // depending on the used Gitlab instance
+        const retryAfter = response.headers.get('Retry-After');
+        const delay = retryAfter
+          ? parseInt(retryAfter, 10) * 1000
+          : Math.min(500 * Math.pow(2, retries - 1), 10000); // Exponential backoff, cap at 10 seconds
+
+        await new Promise(resolve => setTimeout(resolve, delay));
+      }
+      this.queue.pop();
+      // This should never be reached due to the loop condition, but TypeScript needs it
+      throw new Error('Max retries exceeded');
```

Comment:
You can get around this by having `let response: Response` outside of the loop and returning it here. It'd be much cleaner; then you don't need to do the check inside the loop